components-compose/src/main/java/uk/gov/hmrc/components/compose/molecule/selectrow/SelectRowView.kt

```diff
@@ -0,0 +1,125 @@
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.selection.selectable
+import androidx.compose.material.ripple.rememberRipple
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconToggleButton
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.semantics.Role
+import uk.gov.hmrc.components.compose.R
+import uk.gov.hmrc.components.compose.ui.theme.HmrcTheme
+
+object SelectRowView {
+
+    const val NO_ROW_SELECTED_POSITION = -1
+    const val FIRST_ROW_SELECTED_POSITION = 0
+
+    /**
+     * Composable function that creates a SelectRowView with icons and text.
+     *
+     * @param selectRowViewItems List of items to be displayed in the selectable rows.
+     * @param checkedIcon Resource ID for the checked icon.
+     * @param uncheckedIcon Resource ID for the unchecked icon.
+     * @param rowSelectedPosition Initial selection position of a SelectRowView
+     * @param errorText Error text to be displayed, if any.
+     * @param onRowSelected Callback function triggered when an item is clicked.
+     */
+    @Composable
+    operator fun invoke(
+        modifier: Modifier = Modifier,
+        selectRowViewItems: List<String>,
+        checkedIcon: Int = R.drawable.components_select_row_circle_checked,
+        uncheckedIcon: Int = R.drawable.components_select_row_circle_unchecked,
+        rowSelectedPosition: Int,
+        errorText: String = "",
+        onRowSelected: (position: Int, value: String) -> Unit
+    ) {
+        Column(modifier = modifier) {
+            if (errorText.isNotEmpty()) {
+                Text(
+                    text = errorText,
+                    style = HmrcTheme.typography.body.copy(color = HmrcTheme.colors.hmrcRed),
+                    modifier = Modifier.offset(HmrcTheme.dimensions.hmrcSpacing8)
+                )
+                Spacer(modifier = Modifier.height(HmrcTheme.dimensions.hmrcSpacing8))
+            }
+            selectRowViewItems.forEachIndexed { index, selectRow ->
+                Row(
+                    verticalAlignment = Alignment.CenterVertically,
+                    modifier = Modifier
+                        .clickable(
+                            interactionSource = MutableInteractionSource(),
+                            indication = rememberRipple(bounded = false),
+                            onClick = {}
+                        )
+                        .selectable(
+                            selected = (rowSelectedPosition == index),
+                            onClick = {
+                                onRowSelected(index, selectRow)
+                            },
+                            role = Role.RadioButton,
+                        )
```

Comment:
Do we need both `clickable` and `selectable` here? I think `selectable` is an enhanced version of `clickable`.